docs/feature-requests/005-blank-slate.md

```diff
@@ -0,0 +1,265 @@
+
+### Local repository, initialized, no dotcom remotes
+
+We detect this state when the active repository is present but has no dotcom remotes.
+
+<img width="400" alt="github tab, local repository with no GitHub remotes" src="https://user-images.githubusercontent.com/17565/57078536-72f7ba80-6cbc-11e9-84e8-8c2384caccdc.png">
+
+
+#### ...no dotcom repository
+
+_Scenario:_ A user has begun a project locally and now wishes to share it on GitHub.
+
+Clicking the "Create GitHub repository" button opens the [Create repository dialog](#create-repository-dialog) panel in "local repository" mode.
+
+#### ...existing dotcom repository
+
+_Scenario:_ A user has a project cloned locally and it exists on GitHub.
+
+_Scenario:_ A user has a project cloned locally and it exists on GitHub, but they cannot push. Maybe they cloned the project before to read its source more easily, but now they wish to contribute.
+
+Clicking the "Connect GitHub repository" button opens the [Connect repository dialog](#connect-repository-dialog).
+
+### Local repository, initialized, dotcom remotes
+
+This is the state we handle now: when an active repository is present and has one or more dotcom remotes.
+
+## Clone repository dialog
+
+The clone repository dialog begins in search mode. As you type within the text input, once more than three characters have been entered, repositories on GitHub matching the entered text appear in the result list below. Repositories may be identified by full clone URL, `owner/name` pair, or a unique substring of `owner/name`.
+
+<img width="650" alt="clone dialog, empty search" src="https://user-images.githubusercontent.com/17565/57096303-f5937080-6ce2-11e9-8d71-e63677dcba7c.png">
+
+<img width="650" alt="clone dialog, search results" src="https://user-images.githubusercontent.com/17565/57096339-07751380-6ce3-11e9-92ab-4714d6c9df8a.png">
+
+### GitHub clone mode
+
+Clicking on an entry in the search result list or entering the full clone URL of a GitHub repository changes the dialog to "GitHub clone" mode:
+
+<img width="650" alt="clone dialog, GitHub mode" src="https://user-images.githubusercontent.com/17565/57244555-ac0a9480-7006-11e9-8048-950e58f038a3.png">
+
+The "protocol" toggle is initialized to match the value of the `github.preferredRemoteProtocol` config setting. If the protocol is changed, the setting is changed to match.
+
+If the authenticated user cannot push to the chosen GitHub repository, the "fork" checkbox is pre-checked.
+
+While the "fork" checkbox is checked, the "fork destination" dropdown is also shown, populated with organizations to which the user belongs. Organizations within which the user does not has permission to create repositories are disabled with an explanatory suffix. The user's account is selected by default in the "fork destination" dropdown. The "upstream remote name" input is shown and populated with the value of the Atom config setting `github.upstreamRemoteName`.
```

Comment:
I love that this proposal handles and simplifies the process of forking a repo. Repo forking is probably one of the more convoluted git processes and is a huge barrier of entry for people that want to contribute to open source projects. While I would love to see the forking workflow made simpler and more accessible, it is also a complicated enough concept to warrant some more case studies, research, and discussion in order to get it "right". That said, I would like to propose to consider the forking workflow as out-of-scope for the first iteration.

Reply:
I went for it in here because we had feedback from one of our UXR studies that the user bounced from the GitHub tab entirely more or less because we don't handle forking well. But this is totally correct:

> [..] it is also a complicated enough concept to warrant some more case studies, research, and discussion in order to get it "right"

Here's what I'm thinking:

- [ ] I'll re-branch off of my prototype branch for this FR and isolate the dialogs that have the existing fork models.
- [ ] I can re-screenshot and update the write-ups for the dialogs here with forking controls omitted.
- [ ] I can propose "fork controls" as a research topic for our next round of UXR. In our next sprint, I'll write up a script and get up to speed on the way our UXR pipeline works.

Reply:
Coming back to this now, forking definitely adds significant complexity. And, thinking about it more, to do it right we'd likely want to let users create, configure, and push to a fork the first time a push fails... probably in addition to on clone. In any case: punting to lock down scope.